server/src/test/java/mateusz/michal/chat/Unit/Registration/Service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,285 @@
+        }
+
+        @Test
+        @DisplayName("Password short test")
+        public void passworShortTest(){
+            //given
+            UserDTO userDto = UserDTO.builder().name("Maciek123").email("Maciek123@gmail.com").password("Al3$").build();
+            List<MyError> expected = Arrays.asList(MyError.builder().status(422).code(RegisterFormErrorCode.SHORT_PASSWORD)
+                    .title("password is too short").build());
+
+            //when
+            List<MyError> actual = registrationService.validateRequest(userDto);
+
+            //then
+            assertIterableEquals(expected,actual);
+        }
+
+        @Test
+        @DisplayName("Password long test")
+        public void passworLongTest(){
+            //given
+            UserDTO userDto = UserDTO.builder().name("Maciek123").email("Maciek123@gmail.com").
+                    password("AlamaKota#2132gfdge").build();
+            List<MyError> expected = Arrays.asList(MyError.builder().status(422).code(RegisterFormErrorCode.LONG_PASSWORD)
+                    .title("password is too long").build());
+
+            //when
+            List<MyError> actual = registrationService.validateRequest(userDto);
+
+            //then
+            assertIterableEquals(expected,actual);
+        }
+
+        @Test
+        @DisplayName("Password weak and short test")
+        public void passwordWeakAndShortTest(){
+            //given
+            UserDTO userDTO = UserDTO.builder().name("Maciek123").email("Maciek123@gmail.com").password("A").build();
+            List<MyError> expected = Arrays.asList(
+                    MyError.builder().status(422).code(RegisterFormErrorCode.WEAK_PASSWORD)
+                            .title("password is too weak").build(),
+                    MyError.builder().status(422).code(RegisterFormErrorCode.SHORT_PASSWORD)
+                            .title("password is too short").build());
+
+            //when
+            List<MyError> actual = registrationService.validateRequest(userDTO);
+
+            //then
+            assertIterableEquals(expected,actual);
+        }
+
+        @Test
+        @DisplayName("Password weak and long test")
+        public void passwordWeakAndLongTest(){
+            //given
+            UserDTO userDTO = UserDTO.builder().name("Maciek123").email("Maciek123@gmail.com").
+                    password("AlaMaKota12321321321312").build();
+            List<MyError> expected = Arrays.asList(
+                    MyError.builder().status(422).code(RegisterFormErrorCode.WEAK_PASSWORD)
+                            .title("password is too weak").build(),
+                    MyError.builder().status(422).code(RegisterFormErrorCode.LONG_PASSWORD)
+                            .title("password is too long").build());
+
+            //when
+            List<MyError> actual = registrationService.validateRequest(userDTO);
+
+            //then
+            assertIterableEquals(expected,actual);
+        }
+    }
+}
```

Comment:
Brakuje pozytywnych case'ów w testach - poprawny email, poprawne hasło etc

Reply:
Zapomniałem o nich:/ 
Jakkolwiek testów do przypadków łączonych specjalnie nie dodałem. 
To akurat o tych poprawnych kompletnie zapomniałem.